### PR TITLE
fixed #64 added OPML generator

### LINKED ISSUE #64: Add OPML endpoint export for feed list
In #59 we're adding the ability to ingest our current feed list from the wiki.  In #63 we're adding a web server.  It might be nice to connect both of these and offer a way to get the feed list *out* of our app as an [OPML](https://en.wikipedia.org/wiki/OPML) file.  OPML seems to be the standard way of sharing a feed list these days.

What I envision is adding a route that allows for obtaining an XML file in OMPL format with all our student feeds included.  We might be able to use something like https://www.npmjs.com/package/@kohlmannj/opml-generator to do it.

### PR DESCRIPTION
**Pull request by jrdnlx**
Added a js file that has an opml generator to generate as an opml file to fix issue #64 
I had to export the objArray array is wiki-parser.js to access the rss urls and the processedPosts array to access the posts url and posts title. 

I am not sure if this is correctly so please correct me if i did this incorrectly.

### CODE REVIEW

**Comment by Grommers00** [comment]
As soon as this passes the ci/riccleCI and travis build, I'll take a look, but looks like you need to run it through an ESL lint before pushing.

**Comment by humphd** [comment]
@jrdnlx this is getting further out of date.  We should figure out how to get it moving forward and merged.  Can we discuss on Thursday in the lab?

**Comment by jrdnlx** [comment]
Yes sorry, I would love to, or maybe your office hours

**Comment by humphd** [comment]
This is still outstanding, can we get it finished soon?

**Comment by jrdnlx** [comment]
@humphd its failed a check that is in the dependency that has been provided. Thoughts?
 
image

File: src/opml-generator.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+const outlines = {
+  text: 'txt',
+  title: 'My Open Source Experience',
+  type: 'rss',
+  xmlUrl: 'https://jrdnlxopensource.blogspot.com/feeds/posts/default/-/open-source',
+  htmlUrl: 'https://jrdnlxopensource.blogspot.com/',
+};
+
+module.exports = opml(header, outlines);
```
Comment (humphd) [inline src/opml-generator.js]: `outlines` is expected to be an Array vs. an Object, see https://www.npmjs.com/package/opml-generator#usage

File: src/backend/web/routes/opml.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const { XMLHttpRequest } = require('xmlhttprequest');
+const express = require('express');
+const opml = require('../../../opml-generator.js');
+
+const router = express.Router();
+
+const xhr = new XMLHttpRequest();
```
Comment (humphd) [inline src/backend/web/routes/opml.js]: This section can be swapped out for a route, something like this:

```js
router.get('/opml', (req, res) => {
  // call the opml() function here, and return it on res, using the correct content-type of " text/x-opml"
});
```

File: src/backend/web/routes/opml.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+xhr.open('GET', opml);
+xhr.responseType = 'document';
+xhr.send();
+
+module.exports = xhr;
```
Comment (humphd) [inline src/backend/web/routes/opml.js]: You can get rid of this.

File: src/opml-generator.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+const opml = require('opml-generator');
```
Comment (humphd) [inline src/opml-generator.js]: I would move all this code into your opml.js route.  I don't think there's much value breaking it out here.

File: src/backend/web/routes/opml.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    type: 'rss',
+    xmlUrl: 'https://jrdnlxopensource.blogspot.com/feeds/posts/default/-/open-source',
+    htmlUrl: 'https://jrdnlxopensource.blogspot.com/',
+  },
+];
+
+router.get('/opml', (req, res) => {
+  // call the opml() function here, and return it on res, using the correct content-type of " text/x-opml"
+  opml(header, outlines);
+  res.setHeader('Content-type', 'text/x-opml');
```
Comment (birtony) [inline src/backend/web/routes/opml.js]: Right now the return of `opml` is not used anywhere. You need to send it back with the `res`. I assume, you could use [res.send](https://sailsjs.com/documentation/reference/response-res/res-send) method for that.

File: src/backend/web/routes/opml.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  ownerName: 'Jordan',
+};
+
+const outlines = [
+  {
+    text: 'txt',
+    title: 'My Open Source Experience',
+    type: 'rss',
+    xmlUrl: 'https://jrdnlxopensource.blogspot.com/feeds/posts/default/-/open-source',
+    htmlUrl: 'https://jrdnlxopensource.blogspot.com/',
```
Comment (birtony) [inline src/backend/web/routes/opml.js]: According to the [documentation](https://github.com/azu/opml-generator#usage), it seems that `xmlUrl` and `htmlUrl` should be used with quotes:
```js
'xmlUrl': 'https://jrdnlxopensource.blogspot.com/feeds/posts/default/-/open-source',
'htmlUrl': 'https://jrdnlxopensource.blogspot.com/',
```

**Comment by jrdnlx** [comment]
@humphd I think I fixed it, I ran it locally and it works. Please double check

Thanks

**Pull request merged**